internal/worldcomm/worldcomm.go

```diff
@@ -69,7 +62,7 @@ func (p *peer) SetIsAuthenticated(isAuthenticated bool) {
 	p.isAuthenticated = isAuthenticated
 }
 
-type PeersIndex = map[string]map[string]bool
+type PeersIndex = map[string]map[string]struct{}
```

Comment:
what about map[string]*peer? The peer.Topics field is used by the main routine

Reply:
a) subscriptions field is actually a subset of this information, let's remove one
b) we don't need peer information, just a topic count. Since the peer knows their own topics (and then, his state on this structure), we should be able to update the count without addiotional information